content/go/event_log.go.md

```diff
@@ -0,0 +1,205 @@
+objectFilter={"userType":{"operation":"SYSTEM"}}&
+objectMask=mask[userType,eventName,eventCreateDate]&
+resultLimit=0,50
+ */
+func getSystemLogs(sess *session.Session){
+	filter := filter.Build(filter.Path("userType").Eq("SYSTEM"))
+	mask:="userType,eventName,eventCreateDate"
+	systemEvents:=getAllEvents(sess,filter,mask)
+	printLogs(systemEvents)
+}
+
+/**
+Shows the Login Logs
+RequestURL:
+GET https://api.softlayer.com/rest/v3.1/SoftLayer_Event_Log/getAllObjects.json?
+objectFilter={"eventName":{"operation":"^=+Login"}}&
+objectMask=mask[userType;eventName,eventCreateDate]&
+resultLimit=0,50
+ */
+func getLoginLogs(sess *session.Session){
+	filter := filter.Build(filter.Path("eventName").Eq("^= Login"))
+	mask:="userType;eventName,eventCreateDate"
+	eventLogs :=getAllEvents(sess,filter,mask)
+	printLogs(eventLogs)
+}
+
+/**
+Shows the recent Logs
+Request URL:
+GET https://api.softlayer.com/rest/v3.1/SoftLayer_Event_Log/getAllObjects.json?
+objectFilter={"eventCreateDate":{"operation":"greaterThanDate",
+	"options":[{"name":"date","value":["2021-04-24T00:00:00.0000-06:00"]}]}}&
+objectMask=mask[userType;eventName,eventCreateDate]&
+resultLimit=0,50
+ */
+func getRecentLogs(sess *session.Session){
+	offsetDate :=getDateString(-10)
+	filterObject:=filter.Build(filter.Path("eventCreateDate").DateAfter(offsetDate))
+	mask:="userType;eventName,eventCreateDate"
+	recentEvents:=getAllEvents(sess, filterObject,mask)
+	printLogs(recentEvents)
+}
+
+/**
+Pages through all results from the Event_Log. This might take long time.
+*/
+func getAllEvents(
+	sess *session.Session,
+	filter string,
+	mask string,
+) (resp []datatypes.Event_Log) {
+	limit:=50
+	offset:=0
+	eventsSize:=limit
+	var allEvents []datatypes.Event_Log
+	for limit==eventsSize {
+		events := getAllObjects(sess, limit,offset,filter,mask)
+		allEvents=append(allEvents, events...)
+		eventsSize=len(events)
+		eventsSize=0 // comment this line to get all events
+	}
+	return allEvents
+}
+
+/**
+Request URL:
+GET https://api.softlayer.com/rest/v3.1/SoftLayer_Event_Log/getAllObjects.json?
+objectFilter= filter &
+objectMask= mask &
+resultLimit=0,50
+ */
+func getAllObjects(
+	sess *session.Session,
+	limit int,
+	offset int,
+	filter string,
+	mask string,
+) (resp []datatypes.Event_Log) {
+	var service = services.GetEventLogService(sess)
+	result, err := service.
+		Limit(limit).
+		Offset(offset).
+		Filter(filter).
+		Mask(mask).
+		GetAllObjects()
+	if err != nil {
+		fmt.Printf("\n Unable to get Events:\n - %s\n", err)
+		return
+	}
+
+	return result
+}
+
+/**
+ Shows the Event Logs Names.
+ */
+func getEventNames(sess *session.Session){
+	var service = services.GetEventLogService(sess)
+	result, err := service.GetAllEventObjectNames()
+	if err != nil {
+		fmt.Printf("\n Unable to get Events:\n - %s\n", err)
+		return
+	}
+	printAsJsonFormat(result)
+}
+
+/**
+ Shows the Event Logs User Types.
+ */
+func getEventUserTypes(	sess *session.Session) {
```

Comment:
It would be nice to have a method so that the client can filter the event logs for a specific user type.